.github/workflows/test.yml

```diff
@@ -0,0 +1,63 @@
+name: "integration-test"
+on:
+  pull_request:
+  push:
+    branches:
+      - main
+      - 'releases/*'
+
+jobs:
+  test:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v2
+    - uses: actions/setup-python@v2
+      with:
+        python-version: "3.x"
+    - name: Run synapse
+      uses: michaelkaye/setup-matrix-synapse@v0.4.0
+      with: 
+        uploadLogs: true
+        httpPort: 8008
+        customModules: "synapse-s3-storage-provider"
+        customConfig: |
+          registration_shared_secret: QuadraticPeach
+          media_storage_providers:
+            - module: s3_storage_provider.S3StorageProviderBackend
+              store_local: True
+              store_remote: True
+              store_synchronous: True
+              config:
+                bucket: s3-storage-provider-tester
+                endpoint_url: http://127.0.0.1:9000/
+                access_key_id: access_key_id
+                secret_access_key: secret_access_key
+    - name: Setup minio
+      run: |
+        docker run -d -p 9000:9000 --name minio \
+                   -e "MINIO_ACCESS_KEY=access_key_id" \
+                   -e "MINIO_SECRET_KEY=secret_access_key" \
+                   -v /tmp/data:/data \
+                   -v /tmp/config:/root/.minio \
+                   minio/minio server /data
+        export AWS_ACCESS_KEY_ID=access_key_id
+        export AWS_SECRET_ACCESS_KEY=secret_access_key
```

Comment:
A slight issue I have with this is that it took me a few reads to realise that `access_key_id` and `secret_access_key` are actually just string and not some variable that's been generated earlier in the process (I know there's no syntax marker implying that they should be variables, but the naming naturally leads me to think they are). I'd suggest renaming into something like `some_key_id` and `some_secret_key` to make it clearer they are dummy values. Probably not worth blocking merging on this though.